github_workflow.md

```diff
@@ -0,0 +1,153 @@
+label schema/taxonomy.
+
+First, every label should have one of four prefixes:
+
+1. `type:`
+2. `component:`
+3. `cat:`
+4. `status:`
+
+All issues of a given prefix should have the same color to make visual
+identification easy.
+
+### `type:`
+
+The `type:` prefix indicates the type of issue or pull request. There are four
+types:
+
+* `type:bug`
+* `type:enhancement`
+* `type:maintenance`
+* `type:question`
+
+Guidelines:
+
+* Each issue or pull request should have only one `type:` prefixed label.
+* Limiting ourselves to these four types allows us to query our issues in a 
+  structural and relational manner.
+
+### `component:`
+
+The `component:` prefix indicates the component or part of the software the
+issue is related to. Thus, on the JupyterLab repository, we have labels such as:
+
+* `component:About`
+* `component:Application`
+* `component:Build System`
+* `component:Command Palette`
+* `component:Completer`
+* `component:Console`
+* `component:Notebook`
+* `component:Terminal`
+* `component:File Browser`
+* `component:File Editor`
+* ...
+
+Guidelines:
+
+* A single issue or pull request can have multiple `component:` labels, but
+  usually it will be a small number.
+* Each subproject will have its own unique set of `component:` labels.
+
+### `cat:`
+
+The `cat:` prefix indicates the "category" of the issue. The category is sort of
+mixin to the `component:` label to indicate cross-cutting categories such as:
+
+* `cat:Accesibility`
+* `cat:Design and UX`
+* `cat:Documentation`
+* `cat:Examples`
+* `cat:New Contributor`
+* `cat:Settings`
+
+Guidelines:
+
+* A single issue or pull request can have multiple `cat:` labels.
+* Each subproject will have its own unique set of `cat:` labels, but effort
+  should be taken to use common names.
+
+### `status:`
+
+The `status:` prefix indicates the status of the issue or pull request, such as:
+
+* `status:Blocked`
+* `status:Duplicate`
+* `status:Help Wanted`
+* `status:Inactive`
+* `status:Needs Discussion`
+* `status:Needs Info`
+* `status:Needs Review`
+* `status:Needs User Testing`
+* `status:Wont Fix`
+* `status:Work in Progress`
+
+Guidelines:
+
+* A single issue or pull request can have multiple `status:` labels.
+* Subprojects should use ths standard set of `status:` labels, but are free to
+  add additional ones.
+
+## Structured queries 
+
+This taxonomy has been designed to allow structured queries to be performed on
+the issues of one or more repositories. An example of this type of query is
+provided below for the JuptyerLab repository. This visualization shows the
+number of issues grouped by type (color), component (x) and milestone (row):
+
+![JupyterLab Issues:](jupyterlab_issues.png)
+
+## Checklist
+
+Here is a brief checklist developers can use to help them follow this workflow:
+
+* Each issue/PR has one milestone or the `status:Needs Info` label.
+* Each issue/PR has exactly one `type:` label.
+* Each issue/PR has a few `component:` labels.
```

Comment:
A guideline might be if the repo has a number of people focusing on a particular component. If most contributors to a repo are reasonably familiar with the whole repo, there isn't a useful distinction. If it's a big project with lots of contributors with different areas of focus (e.g. notebook, ipython, jupyterlab), components are useful for finding "what should I work on?".

Related in general, it might be useful to note what each practice is meant to accomplish, which might be the best indicator of when each piece is appropriate for a given project (Jupyter or otherwise).

E.g.

> milestones help track a summary of all the changes in a given release, and identify what should be backported, and when a release is ready to be made

> component labels help contributors with a particular area of interest and/or expertise find issues they can help resolve

> status labels help people understand whose attention is needed before it can be resolved (e.g. waiting for more bug reproduction info, waiting for UX review, waiting for a decision)

I can't think of a similar description for `type`, because I don't understand how it helps people make a decision or focus their attention, so I hope you can provide that one :)